lib/util/include/hse/util/arch.h

```diff
@@ -162,8 +162,62 @@ get_time_ns(void)
 #define HSE_L1X_ALIGNED
 #endif
 
-size_t memlcp(const void *s1, const void *s2, size_t len);
-size_t memlcpq(const void *s1, const void *s2, size_t len);
+/**
+ * memlcp() - return longest common prefix
+ * @s1:     byte array one
+ * @s2:     byte array two
+ * @len:    max length to compare
+ *
+ * Return: %memlcp compares byte array %s1 to byte array %s2,
+ * returning the maximum length at which they compare identical.
+ */
+static HSE_ALWAYS_INLINE size_t
+memlcp(const void *s1, const void *s2, size_t len)
+{
+    const uint8_t *lhs = s1;
+    const uint8_t *rhs = s2;
+    size_t i = 0;
+
+    while (i < (len & ~7ul)) {
```

Comment:
I'm sure there's a reason you chose not to precompute (len & ~7ul) prior to entering this while loop.. Curious to learn what that is...

Reply:
Heh..  It should only be computed once prior to entering the loop, and I len mustn't change for the second loop.  Perhaps I should make len a const, that seems justifiable in this case :-)